src/ui/public/vis/vis.js

```diff
@@ -81,18 +81,23 @@ export default function VisFactory(Notifier, Private) {
     this.aggs = new AggConfigs(this, state.aggs);
   };
 
-  Vis.prototype.getState = function () {
+  Vis.prototype.getState = function (full) {
     return {
       title: this.title,
       type: this.type.name,
       params: this.params,
-      aggs: this.aggs.map(function (agg) {
-        return agg.toJSON();
-      }).filter(Boolean),
+      aggs: this.aggs
+        .filter(agg => full || agg.enabled)
+        .map(agg => agg.toJSON())
+        .filter(Boolean),
       listeners: this.listeners
     };
   };
 
+  Vis.prototype.getFullState = function () {
```

Comment:
Makes sense. Reversed into:

- `getState` (fully)
- `getEnabledState` (just enabled aggs)
- `getStateInternal` (does this need a leading `_`?)